test/strings/basic.jl

```diff
@@ -623,7 +626,64 @@ end
 
         @test prevind(strs[2], -1) == 0
         @test prevind(strs[2], -1, 1) == 0
+
+        @test prevind(strs[3], -1) == -2
+        @test prevind(strs[3], -1, 1) == -2
+        @test prevind(strs[4], -1) == -2
+        @test prevind(strs[4], -1, 1) == -2
+
+        @test prevind(strs[5], -4) == 0
+        @test prevind(strs[5], -4, 1) == 0
+        @test prevind(strs[6], -4) == 0
+        @test prevind(strs[6], -4, 1) == 0
+
+        @test prevind(strs[7], -4) == -5
+        @test prevind(strs[7], -4, 1) == -5
+
+        @test prevind(strs[8], -4) == 0
+        @test prevind(strs[8], -4, 1) == 0
     end
+
+    let strs = Any["1234567", GenericString("1234567")]
+        for i in 1:2
+            @test prevind(SubString(strs[i], 1, 3), 10) == 3
+            @test prevind(SubString(strs[i], 1, 3), 10, 1) == 3
+            @test prevind(SubString(strs[i], 1, 3), 10, 2) == 2
+            @test prevind(SubString(strs[i], 1, 3), 3) == 2
+            @test prevind(SubString(strs[i], 1, 3), 2) == 1
+            @test prevind(SubString(strs[i], 1, 3), 1) == 0
+            @test prevind(SubString(strs[i], 1, 3), 3, 1) == 2
+            @test prevind(SubString(strs[i], 1, 3), 2, 1) == 1
+            @test prevind(SubString(strs[i], 1, 3), 1, 1) == 0
+            @test prevind(SubString(strs[i], 1, 3), 3, 2) == 1
+            @test prevind(SubString(strs[i], 1, 3), 2, 2) == 0
+
+            @test nextind(SubString(strs[i], 3), -6) == 1
+            @test nextind(SubString(strs[i], 3), -6, 1) == 1
+            @test nextind(SubString(strs[i], 3), -6, 2) == 2
+            @test nextind(SubString(strs[i], 3), 1) == 2
+            @test nextind(SubString(strs[i], 3), 2) == 3
+            @test nextind(SubString(strs[i], 3), 3) == 4
+            @test nextind(SubString(strs[i], 3), 4) == 5
+            @test nextind(SubString(strs[i], 3), 1, 1) == 2
+            @test nextind(SubString(strs[i], 3), 2, 1) == 3
+            @test nextind(SubString(strs[i], 3), 3, 1) == 4
+            @test nextind(SubString(strs[i], 3), 4, 1) == 5
+            @test nextind(SubString(strs[i], 3), 1, 2) == 3
+            @test nextind(SubString(strs[i], 3), 2, 2) == 4
+            @test nextind(SubString(strs[i], 3), 3, 2) == 5
+            @test nextind(SubString(strs[i], 3), 4, 2) == 6
+        end
+    end
+
+    @test prevind(SubString("1234567", 1, 3), 0) == -1
```

Comment:
Why do we return `-1` in that case?

Reply:
because `prevind("1234567", 0) == -1`. It is an inconsistency between `prevind` for `String` and for `AbstractString`.